### PR TITLE
Feat/property controller

### PR DESCRIPTION
**Pull request by OmoT7**
This pull request introduces a new `PropertyController` class with comprehensive business logic for handling property-related actions in a Monopoly-style game, along with full boundary value analysis documentation and a suite of unit tests. The main changes include the implementation of the controller, thorough documentation of its behavior, and extensive tests to ensure correctness across various scenarios.

**Key changes:**

### Controller Implementation

* Added the `PropertyController` class in `PropertyController.java`, providing methods for property purchase prompts, buying, declining, rent payments, and forced sales, with input validation and business logic for each scenario.

### Documentation

* Created `PropertyController BVA Analysis` documentation in `PropertyController.md`, detailing boundary value analysis and expected outcomes for all controller methods and edge cases.

### Testing

* Added `PropertyControllerTests.java`, implementing unit tests for all controller methods, covering null checks, business logic branches, and edge cases as described in the BVA documentation.

### CODE REVIEW

**Review by jskpaul — COMMENTED** [review]
I'm not sure if it's fine to just refactor every test into nonnull at once, since it's technically violating TDD. But my gut reaction is that these need to be recommitted into individual test cases
UPDATE: Not a problem when refactoring

File: src/test/java/controller/PropertyControllerTests.java [hunk 1]
```diff
@@ -311,4 +311,27 @@ public void TC21_HandleForcedSale_NoProperties_ReturnsFalse() {
...
+
+    @Test
+    public void TC22_HandleForcedSale_OnePropertyCoversAmount_ReturnsTrue() {
+        PropertyController controller = new PropertyController();
+        Player player = EasyMock.createMock(Player.class);
+        Property property = EasyMock.createMock(Property.class);
+        Set<Property> properties = new HashSet<>();
+        properties.add(property);
+
+        EasyMock.expect(player.canAfford(300.0)).andReturn(false);
```
Comment (jskpaul) [inline src/test/java/controller/PropertyControllerTests.java:323]: Technically magic number? I see more instances of it below, but idk if they're fine for test files

**Review by jskpaul — APPROVED** [review]
